ratatool-examples/src/main/scala/com/spotify/ratatool/examples/scalacheck/GenExample.scala

```diff
@@ -151,4 +173,20 @@ object ExampleTableRowGen {
        */
       .tryAmend(intListGen)(_.getRecord("nullable_record").set("repeated_int_field"))
       .tryAmend(freqGen)(_.getRecord("nullable_record").set("frequency_string_field"))
+
+  private val recordIdGen: Gen[String] = Gen.alphaUpperStr
+
+  private val tableRowGenDup: Gen[TableRow] =
+    tableRowOf(tableSchema)
+
+  val exampleRecordGen2: Gen[(TableRow, TableRow)] =
+    (tableRowGen, tableRowGenDup)
+      .tupled
+      .amend2(recordIdGen)(_.set("record_id"), _.set("record_id"))
+
+  val correlatedRecordGen: Gen[(TableRow, TableRow)] =
```

Comment:
Can call a`.map` after to set the nested field so that the output is a single TableRow with the record ID fields correlated.

Reply:
Same for Avro